### PR TITLE
Use Coil

### LINKED ISSUE #4193: Use Coil for image loading
We currently use Glide, but Coil seems to offer better performance based on anecdotal feedback from J2K/Neko users in the context of viewing GIFs.

### PR DESCRIPTION
**Pull request by ivaniskandar**
Closes #4193

Custom cover and proper caching is still broken, need to wrap the previous ModelLoader logic around my head first.

### CODE REVIEW

**Comment by arkon** [comment]
FYI, you could refer to [TachiyomiJ2K](https://github.com/Jays2Kings/tachiyomiJ2K)'s codebase too, since it already uses Coil.

File: app/build.gradle.kts [hunk 1]
```diff
@@ -203,6 +203,11 @@ dependencies {
     implementation("com.github.bumptech.glide:okhttp3-integration:$glideVersion")
     kapt("com.github.bumptech.glide:compiler:$glideVersion")
 
+    val coilVersion = "1.2.0"
+    implementation("io.coil-kt:coil:$coilVersion")
+    implementation("io.coil-kt:coil-gif:$coilVersion")
+    implementation("com.github.Commit451.coil-transformations:transformations:1.0.0")
```
Comment (CarlosEsco) [inline app/build.gradle.kts:209]: Are you sure a transformer library is needed?   Coil already comes with a rounded corner transformer, and the center crop transformer shouldnt be needed (we dont use one on neko/j2k)

Comment (CarlosEsco) [reply app/build.gradle.kts:209]: also if you for some reason do need a transformer i tend to prefer
https://github.com/wasabeef/transformers

Comment (ivaniskandar) [reply app/build.gradle.kts:209]: Hmmm you're right, looks like setting `android:scaleType="centerCrop"` to the ImageView is enough. I'll remove it.

**Comment by ivaniskandar** [comment]
Custom cover and files caching should work now.

One thing is I'm not sure how to test the automatic cover refresh functions, I assume it works as long as the user restarted the app after the library finish updating to clear the MemoryCache.

File: app/src/main/java/eu/kanade/tachiyomi/App.kt [hunk 2]
```diff
@@ -64,6 +74,23 @@ open class App : Application(), LifecycleObserver {
...
+                    add(ImageDecoderDecoder(this@App))
+                } else {
+                    add(GifDecoder())
+                }
+                add(ByteBufferFetcher())
+                add(MangaCoverFetcher())
+            }
+            okHttpClient(Injekt.get<NetworkHelper>().coilClient)
+            crossfade(300)
+            allowRgb565(getSystemService<ActivityManager>()!!.isLowRamDevice)
```
[the remark this answers is not in the document]
Comment (ivaniskandar) [reply app/src/main/java/eu/kanade/tachiyomi/App.kt:90]: Ah well then is it better to tie this options with [trueColor preference](https://github.com/tachiyomiorg/tachiyomi/blob/master/app/src/main/java/eu/kanade/tachiyomi/data/preference/PreferencesHelper.kt#L106)?

[the remark this answers is not in the document]
Comment (arkon) [reply app/src/main/java/eu/kanade/tachiyomi/App.kt:90]: I thought about it a bit more after I wrote that initial comment. My original comment was with the context of the reader, where Coil wouldn't be used as often (i.e. just for GIFs, and likely not for longstrip content where we typically encounter the large image issue), so maybe this is fine. We could come back to it depending on user reports, if any.

File: app/src/main/java/eu/kanade/tachiyomi/ui/browse/extension/ExtensionHolder.kt [hunk 3]
```diff
@@ -41,11 +41,8 @@ class ExtensionHolder(view: View, val adapter: ExtensionAdapter) :
             else -> ""
         }.toUpperCase()
 
-        GlideApp.with(itemView.context).clear(binding.image)
```
[the remark this answers is not in the document]
Comment (ivaniskandar) [reply app/src/main/java/eu/kanade/tachiyomi/ui/browse/extension/ExtensionHolder.kt:44]: Oops yes I forgot about that

**Review by arkon — APPROVED** [review]
Thanks again!

**Pull request merged**